src/ipinfoWrapper.ts

```diff
@@ -72,32 +84,46 @@ export default class IPinfoWrapper {
                         data += chunk;
                     });
 
-                    res.on("close", () => {
-                        const ipinfo: IPinfo = JSON.parse(data);
+                    if (!this._isStatusCodeIn400Range(res.statusCode)) {
```

Comment:
So `res.statusCode` is actually available at this point, when `https.request` calls this callback? It makes sense to me actually, but just confirming.

Reply:
@UmanShahzad I think status code should be available to the `res`(response) object returned in callback, as in example on official page at [Making HTTP requests with Node.js](https://nodejs.dev/learn/making-http-requests-with-nodejs). And it'll only be called once, so that seems to be the only chance where we can get the status code.
Also we can register the "response" event listener. Both seems to be called same way, as also mentioned in the [doc](https://nodejs.org/api/http.html#httprequesturl-options-callback) :

> "The optional callback parameter will be added as a one-time listener for the ['response'](https://nodejs.org/api/http.html#event-response) event."

Reply:
Alright, as long as it's tested and it actually does work (we can console.log) then all good with me.

Reply:
Tested it with console.log. It is displaying correct status code from callback.